addons/web/static/tests/core/py_js/py_interpreter.test.js

```diff
@@ -214,33 +214,9 @@ describe.tags("headless")("comparisons", () => {
     });
 
     test("mixed types comparisons", () => {
-        expect(evaluateExpr("None < 42")).toBe(true);
-        expect(evaluateExpr("None > 42")).toBe(false);
-        expect(evaluateExpr("42 > None")).toBe(true);
-        expect(evaluateExpr("None < False")).toBe(true);
-        expect(evaluateExpr("None < True")).toBe(true);
-        expect(evaluateExpr("False > None")).toBe(true);
-        expect(evaluateExpr("True > None")).toBe(true);
-        expect(evaluateExpr("None > False")).toBe(false);
-        expect(evaluateExpr("None > True")).toBe(false);
-        expect(evaluateExpr("0 > True")).toBe(false);
-        expect(evaluateExpr("0 < True")).toBe(true);
-        expect(evaluateExpr("1 <= True")).toBe(true);
-        expect(evaluateExpr('False < ""')).toBe(true);
-        expect(evaluateExpr('"" > False')).toBe(true);
-        expect(evaluateExpr('False > ""')).toBe(false);
-        expect(evaluateExpr('0 < ""')).toBe(true);
-        expect(evaluateExpr('"" > 0')).toBe(true);
-        expect(evaluateExpr('0 > ""')).toBe(false);
-        expect(evaluateExpr("3 < True")).toBe(false);
-        expect(evaluateExpr("3 > True")).toBe(true);
-        expect(evaluateExpr("{} > None")).toBe(true);
-        expect(evaluateExpr("{} < None")).toBe(false);
-        expect(evaluateExpr("{} > False")).toBe(true);
-        expect(evaluateExpr("{} < False")).toBe(false);
-        expect(evaluateExpr("3 < 'foo'")).toBe(true);
-        expect(evaluateExpr("'foo' < 4444")).toBe(false);
-        expect(evaluateExpr("{} < []")).toBe(true);
```

Comment:
Add a test case for that to show the type exception when the datatype is different.